### DISCUSSION TITLE
[Question] 关于@opensumi/playwright使用的一些问题寻求帮助

### DISCUSSION THREAD

**Discussion by 767746649zyy** [opensumi/core#2165] [General]
首先我clone了ide-electron项目，并在根目录加入tools/playwright文件夹内容（内容同opensumi/core下的playwright文件内容相同），以及加入了与之相关的configs文件，运行自动化测试命令 pnpm run ui-tests 运行报错如下
![image](https://user-images.githubusercontent.com/23164065/210720582-702abe8d-02a3-4654-8870-5573cb3724e2.png)

问题1：目前@opensumi/playwright是否仅支持web项目，不支持electron项目？
问题2：使用@opensumi/playwright时，一定是执行pack构建完成的项目才可以执行自动化测试吗？

**Comment by erha19** [comment]
@767746649zyy 目前 `@opensumi/playwright` 包编写是基于 Web 环境的测试编写的，不能直接在 Electron 项目内使用，有兴趣可以自行实现，重点在于 `OpenSumiApp.load` 及 https://github.com/opensumi/core/blob/c26945d8f9fe6d524a1db5d5d007d1ad06467a38/tools/playwright/src/tests/hooks/index.ts 需要额外的修改配置。

使用 Playwright 测试 Electron 项目并不一定是需要将产物构建完，对于能用 `electron .` 启动的产物也是可以进行自动化测试的。
